Modules/Registration/Common/test/itkMeanSquaresImageMetricTest.cxx

```diff
@@ -295,6 +298,110 @@ itkMeanSquaresImageMetricTest(int, char *[])
   //-------------------------------------------------------
   metric->Print(std::cout);
 
+
+   //-------------------------------------------------------
+  // Check consistency between BSplineTransform
+  // derivatives computed with vs omitting weights caching.
+  //-------------------------------------------------------
+
+  // Define types/constants
+  constexpr unsigned int splineOrder = 3;
+  constexpr unsigned int nodesPerDimension = 8;
+  using BSplineTransformType = itk::BSplineTransform<CoordinateRepresentationType, ImageDimension, splineOrder>;
+  using InitializerType = itk::BSplineTransformInitializer<BSplineTransformType, MovingImageType>;
+
+  // Offset moving image to ensure derivatives for BSplineTransform are non-zero.
+  MovingImageType::PointValueType bSplineTestMovingImageOrigin[] = { 1.0f, 1.0f };
+  movingImage->SetOrigin(bSplineTestMovingImageOrigin);
+
+  // Initialise BSplineTransform
+  auto bSplineTransform = BSplineTransformType::New();
+  auto initializer = InitializerType::New();
+  BSplineTransformType::MeshSizeType meshSize;
+  meshSize.Fill(nodesPerDimension - splineOrder);
+  initializer->SetTransform(bSplineTransform);
+  initializer->SetImage(movingImage);
+  initializer->SetTransformDomainMeshSize(meshSize);
+  initializer->InitializeTransform();
+
+  // Set bSplineTransform parameters
+  ParametersType bSplineParameters(bSplineTransform->GetNumberOfParameters());
+  bSplineParameters.Fill(0.0);
+  bSplineTransform->SetParameters(bSplineParameters);
+
+  // Initialise metric without caching
+  auto metricNoCaching = MetricType::New();
+  metricNoCaching->SetFixedImage(fixedImage);
+  metricNoCaching->SetMovingImage(movingImage);
+  metricNoCaching->SetInterpolator(interpolator);
+  metricNoCaching->SetFixedImageRegion(fixedImage->GetBufferedRegion());
+  metricNoCaching->SetTransform(bSplineTransform);
+  metricNoCaching->SetUseCachingOfBSplineWeights(false);
+  try
+  {
+    metricNoCaching->Initialize();
+  }
+  catch (const itk::ExceptionObject & e)
+  {
+    std::cout << "Failed to initialise metric to test BSplineTransform derivative computation omitting weights caching."
+              << std::endl;
+    std::cout << "Reason " << e.GetDescription() << std::endl;
+
+    return EXIT_FAILURE;
+  }
```

Comment:
Use `ITK_TRY_EXPECT_NO_EXCEPTION( metricNoCaching->Initialize());` instead of https://github.com/InsightSoftwareConsortium/ITK/pull/4982/files#diff-c3d685418473012ef3d09730cf43126a837aba1b9e8cc7e63cc1cd09616c0239R342-R351

Reply:
Thank you for this suggestion! The metricNoCaching and MetricWithCaching are being combined into a single metric following another suggestion - ITK_TRY_EXCEPT_NO_EXCEPTION is used there.

Reply:
I think you need to `#include "itkTestingMacros.h"`